plugin/formatting.py

```diff
@@ -24,13 +24,13 @@ class WillSaveWaitTask(SaveTask):
     def is_applicable(cls, view: sublime.View) -> bool:
         return bool(view.file_name())
 
-    def __init__(self, view: sublime.View, on_complete: Callable[[], None]) -> None:
-        super().__init__(view, on_complete)
+    def __init__(self, task_runner: LspTextCommand, on_complete: Callable[[], None]) -> None:
+        super().__init__(task_runner, on_complete)
         self._session_iterator = None  # type: Optional[Iterator[Session]]
 
     def run_async(self) -> None:
         super().run_async()
-        self._session_iterator = sessions_for_view(self._view, 'textDocumentSync.willSaveWaitUntil')
+        self._session_iterator = sessions_for_view(self._task_runner.view, 'textDocumentSync.willSaveWaitUntil')
```

Comment:
this can also use `sessions(...)` I believe?

Reply:
type-checking doens't complain :)

Reply:
Also tested on actual server